File: SmartBrowser_java_eclipse/src/com/teuskim/sbrowser/HumanProgressBar.java
package com.teuskim.sbrowser;

import android.content.Context;
import android.os.Handler;
import android.os.Message;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;
import android.widget.RelativeLayout;

public class HumanProgressBar extends RelativeLayout {
	
	private static final int MAX_PROGRESS = 100;
	private ImageView mPushman;
	private View mBtnCloseProgress;
	
	private int mDestProgress;
	private int mCurrProgress;
	private Handler mHandler = new Handler(){

		@Override
		public void handleMessage(Message msg) {
			super.handleMessage(msg);
			
			if(mCurrProgress < mDestProgress){
				mCurrProgress++;
				LayoutParams lp = (LayoutParams) mPushman.getLayoutParams();
				lp.leftMargin = (int)((getWidth() - mPushman.getWidth()) * mCurrProgress / MAX_PROGRESS);
				mPushman.setLayoutParams(lp);
			}
			
			if(mCurrProgress < MAX_PROGRESS-5){
				mHandler.sendEmptyMessage(0);
			}
			else{
				hide();
			}
			
			if(mCurrProgress % 20 >= 10)
				mPushman.setImageResource(R.drawable.img_loading_pushman02);
			else
				mPushman.setImageResource(R.drawable.img_loading_pushman01);
			
		}
		
	};

	public HumanProgressBar(Context context, AttributeSet attrs, int defStyle) {
		super(context, attrs, defStyle);
		init(context);
	}

	public HumanProgressBar(Context context, AttributeSet attrs) {
		super(context, attrs);
		init(context);
	}

	public HumanProgressBar(Context context) {
		super(context);
		init(context);
	}

	private void init(Context context){
		LayoutInflater.from(context).inflate(R.layout.human_progress_bar, this);
		mPushman = (ImageView) findViewById(R.id.pushman);
		mBtnCloseProgress = findViewById(R.id.btn_close_progress);
	}
	
	public void setProgress(int progress){
		
		mDestProgress = progress;
	}
	
	public void setStopListener(OnTouchListener listener){
		mBtnCloseProgress.setOnTouchListener(listener);
	}
	
	public void show(){
		setVisibility(VISIBLE);
		mDestProgress = 0;
		mCurrProgress = 0;
		mHandler.sendEmptyMessage(0);
	}
	
	public void hide(){
		setVisibility(GONE);
	}
}

File: SmartBrowser_kotlin/app/src/main/java/com/teuskim/sbrowser/HumanProgressBar.kt
package com.teuskim.sbrowser

import android.content.Context
import android.os.Handler
import android.os.Message
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import android.widget.RelativeLayout

class HumanProgressBar : RelativeLayout {
    private var mPushman: ImageView? = null
    private var mBtnCloseProgress: View? = null

    private var mDestProgress: Int = 0
    private var mCurrProgress: Int = 0
    private val mHandler = object : Handler() {

        override fun handleMessage(msg: Message) {
            super.handleMessage(msg)

            if (mCurrProgress < mDestProgress) {
                mCurrProgress++
                val lp = mPushman!!.layoutParams as RelativeLayout.LayoutParams
                lp.leftMargin = (width - mPushman!!.width) * mCurrProgress / MAX_PROGRESS
                mPushman!!.layoutParams = lp
            }

            if (mCurrProgress < MAX_PROGRESS - 5) {
                this.sendEmptyMessage(0)
            } else {
                hide()
            }

            if (mCurrProgress % 20 >= 10)
                mPushman!!.setImageResource(R.drawable.img_loading_pushman02)
            else
                mPushman!!.setImageResource(R.drawable.img_loading_pushman01)

        }

    }

    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle) {
        init(context)
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init(context)
    }

    constructor(context: Context) : super(context) {
        init(context)
    }

    private fun init(context: Context) {
        LayoutInflater.from(context).inflate(R.layout.human_progress_bar, this)
        mPushman = findViewById<View>(R.id.pushman) as ImageView
        mBtnCloseProgress = findViewById(R.id.btn_close_progress)
    }

    fun setProgress(progress: Int) {

        mDestProgress = progress
    }

    fun setStopListener(listener: View.OnTouchListener) {
        mBtnCloseProgress!!.setOnTouchListener(listener)
    }

    fun show() {
        visibility = View.VISIBLE
        mDestProgress = 0
        mCurrProgress = 0
        mHandler.sendEmptyMessage(0)
    }

    fun hide() {
        visibility = View.GONE
    }

    companion object {

        private val MAX_PROGRESS = 100
    }
}
